macOS/DuckDuckGo/Favicons/Model/FaviconDownloader.swift

```diff
@@ -0,0 +1,141 @@
+    }
+
+    private func downloadUsingWKDownload(from url: URL, using webView: WKWebView?) async throws -> Data {
+        // Use provided webView (to share session/cookies), or create a temporary one if needed
+        let targetWebView = webView ?? createTemporaryWebView()
+
+        let download = await targetWebView.startDownload(using: URLRequest(url: url))
+        return try await withCheckedThrowingContinuation { continuation in
+            let task = DownloadTask(url: url, continuation: continuation)
+            pendingDownloads[download] = task
+            download.delegate = self
+        }
+    }
+
+    private func createTemporaryWebView() -> WKWebView {
+        let configuration = WKWebViewConfiguration()
+        configuration.websiteDataStore = .nonPersistent()
+        let webView = WKWebView(frame: .zero, configuration: configuration)
+        webView.navigationDelegate = self
+        return webView
+    }
+
+    private struct DownloadTask {
+        let url: URL
+        let continuation: CheckedContinuation<Data, Error>
+        var destinationURL: URL?
+    }
+}
+
+extension FaviconDownloader: WKNavigationDelegate {
+
+    nonisolated func webView(
+        _ webView: WKWebView,
+        decidePolicyFor navigationAction: WKNavigationAction,
+        preferences: WKWebpagePreferences,
+        decisionHandler: @escaping (WKNavigationActionPolicy, WKWebpagePreferences) -> Void
+    ) {
+        // We're only using this WebView for downloads, not navigation
+        decisionHandler(.cancel, preferences)
+    }
+}
+
+extension FaviconDownloader: WKDownloadDelegate {
+
+    func download(
+        _ download: WKDownload,
+        decideDestinationUsing response: URLResponse,
+        suggestedFilename: String
+    ) async -> URL? {
+        // Create a temporary file URL for the download
+        let tempDirectory = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
+        let fileName = UUID().uuidString
+        let destinationURL = tempDirectory.appendingPathComponent(fileName)
+
+        // Store the destination URL so we can read the file later
+        if var task = pendingDownloads[download] {
+            task.destinationURL = destinationURL
+            pendingDownloads[download] = task
+        }
+
+        return destinationURL
+    }
+
+    func downloadDidFinish(_ download: WKDownload) {
+        guard let task = pendingDownloads.removeValue(forKey: download),
+              let destinationURL = task.destinationURL else {
+            Logger.favicons.error("FaviconDownloader: No task or destination found for finished download")
+            return
+        }
+
+        // Read the downloaded file
+        do {
+            let data = try Data(contentsOf: destinationURL)
+            task.continuation.resume(returning: data)
+
+            // Clean up the temporary file
+            try? FileManager.default.removeItem(at: destinationURL)
```

Comment:
Maybe we could do a defer to make sure the file is removed regardless if there was issues reading it?
```
    defer {
        try? FileManager.default.removeItem(at: destinationURL)
    }
```